scripts/install.sh

```diff
@@ -71,59 +90,88 @@ Exec=$BINARY_PATH
 Icon=$HOME/.local/$APP_NAME/lib/$APP_NAME.png
 Terminal=false
 Type=Application
-StartupWMClass=com-abdownloadmanager-desktop-AppKt"
-
-    echo "$desktop_file_content" >"$HOME/.local/share/applications/abdownloadmanager.desktop"
+StartupWMClass=com-abdownloadmanager-desktop-AppKt
+EOF
 }
 
-# Download the latest version of the app
+# --- Download the latest version of the app
 download_zip() {
-    echo "Downloading AB Download Manager ..."
-    url=$(get_download_url)
-    curl -L -o "/tmp/$ASSET_NAME" "$url"
-    echo "Download Finished"
+    # Remove the app tarball if it exists in /tmp
+    rm -f "/tmp/$ASSET_NAME"
+
+    logger "downloading AB Download Manager ..."
+    # Perform the download with curl
+    if curl --progress-bar -fSL -o "/tmp/$ASSET_NAME" "${DOWNLOAD_URL}"; then
+        logger "download finished successfully"
+    else
+        logger error "Download failed! Something Went Wrong"
+        logger error "Hint: Check Your Internet Connectivity"
+        # Optionally remove the partially downloaded file
+        rm -f "/tmp/$ASSET_NAME"
+    fi
 }
 
-# Install the app
+
+# --- Install the app
 install_app() {
-    echo "Installing AB Download Manager ..."
-    mkdir -p "$HOME/.local"
-    tar -xzf "/tmp/$ASSET_NAME" -C "$HOME/.local"
 
-    # Setup ~/.local directories
+    logger "Installing AB Download Manager ..."
+    # --- Setup ~/.local directories
     mkdir -p "$HOME/.local/bin" "$HOME/.local/share/applications"
+    tar -xzf "/tmp/$ASSET_NAME" -C "$HOME/.local"
+
+    # --- remove tarball after installation
+    rm "/tmp/$ASSET_NAME"
 
     # Link the binary to ~/.local/bin
     ln -s "$BINARY_PATH" "$HOME/.local/bin/$APP_NAME"
 
     # Create a .desktop file in ~/.local/share/applications
     generate_desktop_file
 
-    echo "AB Download Manager installed successfully"
-    echo "Open it from your Applications menu or run '$APP_NAME' from the terminal"
-    echo "Make sure $HOME/.local/bin is in your PATH"
+    logger "AB Download Manager installed successfully"
+    logger "it can be found in Applications menu or run '$APP_NAME' in terminal"
+    logger "Making sure $HOME/.local/bin exists in PATH"
+    if ! $(echo "$PATH" | grep "$HOME/.local/bin" >/dev/null 2>&1); then
+        logger "Adding $HOME/.local/bin to ${USER}'s PATH"
+        echo "export $HOME/.local/bin:$PATH" >> $HOME/.bashrc
+    fi
```

Comment:
it can also be optional since we changing the user's profile

Reply:
thought about it, the script can just print the instruction to add in the `.bashrc`, most users search the application UI in the applications menu, so it could be ignored to modify the user's `.bashrc`